### PR TITLE
Add method to allow Commit class to show all files

### PR DESCRIPTION
**Pull request by shahbhavya85**

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @shahbhavya85, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

File: src/Bart/Git/Commit.php [hunk 1]
```diff
@@ -57,6 +57,21 @@ public function message()
 		return $result->getOutput(true);
 	}
 
+	/** 
+	 * @return string List of modified files
+	 */
+
```
Comment (asheepapart) [inline src/Bart/Git/Commit.php]: Please remove this extraneous line.

Is the return type string[] or string? We should be returning an array of strings, one for each file path. 

```
/**
 * @return string[] The files names changed in commit
 */

File: src/Bart/Git/Commit.php [hunk 2]
```diff
@@ -57,6 +57,21 @@ public function message()
...

+	 * @return string List of modified files
+	 */
+
+	public function getFileList()
+	{
+		$result = $this->gitRoot->getCommandResult('show --pretty="format:" --name-only %s', $this->revision);
+
+		if (!$result->wasOk()) {
+			throw new GitException("Could not get contents of commit {$this}");
```
Comment (asheepapart) [inline src/Bart/Git/Commit.php]: `could not get file names from commit {$this}`

File: test/Bart/Git/CommitTest.php [hunk 3]
```diff
@@ -30,6 +30,23 @@ private function stubGitRootForMessage($message)
...
 
+	private function stubGitRootFileList($getFileList)
+	{
+		$getFileList = "commit 31ac1584186257ce6aa14d9cefd78a4b2f89c90e
+Author: Nishad Singh <nsingh@box.com>
+Date:   Mon Aug 10 14:09:36 2015 -0700
+
+conf_override/features.conf
+conf_override/services.conf
+$getFileList";
```
Comment (asheepapart) [inline test/Bart/Git/CommitTest.php]: What's this extra `$getFileList` doing here?

File: test/Bart/Git/CommitTest.php [hunk 4]
```diff
@@ -30,6 +30,23 @@ private function stubGitRootForMessage($message)
...
+
+		$this->gitRoot = $this->shmock('\Bart\Git\GitRoot', function($root) use ($getFileList) {
+			$resultStub = new StubbedCommandResult([$getFileList], 0);
```
Comment (asheepapart) [inline test/Bart/Git/CommitTest.php]: The first parameter to `StubbedCommandResult()` should be an array of the individual lines of the output.

File: test/Bart/Git/CommitTest.php [hunk 5]
```diff
@@ -39,6 +56,15 @@ public function testMessage()
 		$this->assertContains('a57a2664feafb26c61d269babc63b272ed87544d', $commit->message(), 'hash');
 	}
 
+	public function testgetFileList()
```
Comment (asheepapart) [inline test/Bart/Git/CommitTest.php]: Rename to `getGetFileList()`

Please also test for the case when the command fails (i.e. `$result` was not OK).

**Pull request closed without merging**